### ISSUE TITLE
eks with secondary network for pods

### ISSUE THREAD

**Issue by davirezendegb** [aws/amazon-vpc-cni-k8s#3072]
Following the documentation, we end up in the following scenario.
https://aws.github.io/aws-eks-best-practices/networking/custom-networking/

The pod receives the IPs from the secondary subnet and to try to access resources outside the VPC, the main IP of the node in the main network is used, but for resources in the same VPC it tries to use the secondary IP that the pod receives. Is there a way to mask the pod IP to the main IP for internal resources in the same VPC?

pod ip: 100.127.35.58
node ip: 192.168.1.44

when trying to access an RDS resource in the same vpc the pod uses the ip 100.127.35.58

to access a resource outside the vpc a masking is done so that the pod uses the node ip 192.168.1.44

We would like this masking to also be done for internal resources in the same VPC

**Comment by orsenthil** [comment]
> Is there a way to mask the pod IP to the main IP for internal resources in the same VPC?

No, there is not a way to mask the pod ip to the main ip (or node ip) of internal resources in the same VPC. 

If you use the hostnetworking pod, then the ip of the pod is the same as the node's primary ip.

**Comment by davirezendegb** [comment]
do we not have the possibility to create a function for this? 
"AWS_VPC_K8S_CNI_INTERNALSNAT=true"

**Comment by orsenthil** [comment]
Do you mean, create a new flag to have an Internal SNAT? I am not sure how that will work with hostnetwork pods.

**Comment by davirezendegb** [comment]
the intention is to make the pods receive the ips of the secondary subnet as expected, but to communicate internally use the ip of the main subnet that is also tied to the node
![Captura de Tela_Área de Seleção_20241015174143](https://github.com/user-attachments/assets/d3e32585-4268-467f-a119-f877fca46604)
for example, the pods receive the range 100.127.xxx.xxx and intermanete they use this ip 100.127.xxx.xxx to communicate with an RDS.

the idea is to use the primary ip of the node: 10.72.xxx.xxx (this occurs for when the resource is outside the vpc)

**Comment by davirezendegb** [comment]
in GCP and azure there is this way of performing this configuration

https://cloud.google.com/kubernetes-engine/docs/how-to/ip-masquerade-agent

https://github.com/Azure/ip-masq-agent-v2

**Comment by orsenthil** [comment]
@davirezendegb - can you raise this feature request [containers-roadmap](https://github.com/aws/containers-roadmap/). It needs to be evaluated and prioritized as a product requirement for CNI.

**Comment by arehmandev** [comment]
Hi this is a blocker for us too, we want to use a seperate subnet for pods whilst maintaining routing for nodes via internalsnat

**Comment by chlunde** [comment]
With https://github.com/aws/containers-roadmap/issues/1342 - you could effectively also achieve this. Either implementation would be very welcome to reduce the cost of a secure configuration.

**Comment by nd-at-globetel** [comment]
Any update on this? 

Same scenario.

The VPC used by the EKS cluster with VPC CNI Custom Networking enabled and the RDS is the same but different subnets.

I know that the default behavior expected as per official documentation for the SNAT implementation.
